src/pebs.c

```diff
@@ -30,18 +30,21 @@ MEMKIND_EXPORT int pebs_debug_num_samples = 0;
 
 void pebs_monitor()
 {
-    // must call this before read from data head
-    asm volatile("lfence" ::: "memory");
-
     int all_cpu_samples = 0;
 
     for (size_t cpu_idx = 0u; cpu_idx < CPU_LOGICAL_CORES_NUMBER; ++cpu_idx) {
         perf_event_mmap_page_t *pebs_metadata =
             (perf_event_mmap_page_t *)pebs_mmap[cpu_idx];
 
+        if (pebs_metadata == NULL)
```

Comment:
is it ever possible on an initialized thread? if not, maybe an assert should be a better solution?